client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -1180,6 +1148,47 @@ const ConnectedPlanFeatures2023Grid = connect(
 
 		const isJetpackNotAtomic = isJetpack && ! isSiteAT;
 
+		const hasDiscountNotice = () => {
+			const { withDiscount } = ownProps;
+			const bannerContainer = getBannerContainer();
+			if ( ! bannerContainer ) {
+				return false;
+			}
+
+			const activeDiscount = getDiscountByName( withDiscount );
+			if ( ! activeDiscount || hasPlaceholders( planProperties ) || ! canUserPurchasePlan ) {
+				return false;
+			}
+			return true;
+		};
+
+		const higherPlanAvailable = () => {
+			const currentPlan = currentSitePlan?.productSlug;
+			const highestPurchasablePlan = planProperties
+				.filter( ( { planName } ) => planName !== PLAN_ENTERPRISE_GRID_WPCOM )
+				.filter( ( { availableForPurchase } ) => availableForPurchase )
+				.pop();
+			const { planName: highestPurchasablePlanName } = highestPurchasablePlan ?? {};
+			return currentPlan !== highestPurchasablePlanName;
+		};
```

Comment:
Not sure I follow why we moved these into the loop. was it necessary? 🤔

Reply:
I guess you mean the connect function here?
I wanted lift the logic required to deduce the state of `hideCreditNotice` to the connect function. This is because this state needed to be used at 2 different parts of the component. I know I could have just moved it into a function inside the class and called it. However I was planning on doing a refactor of possibly getting rid of this React.portal implementation and consolidating the notice logic higher up in the tree into it's own component. And the logic related to it will be close by instead of in multiple functions inside the above class. Sorry I guess I could have been a bit more clear in my pull request.

Do you recommend reverting to doing all of this inside the class? 🤔